Create R version policy document (#403)
easystats/easystats: vignettes/version_policy.Rmd

improvements, but also sometimes changes the default value of some parameters
(such as switching from `stringsAsFactors = TRUE` to `FALSE` in R 4.0.0).

While there is benefit in supporting older R versions, doing so adds 
maintenance burden, especially in a large ecosystem like **easystats** that
contains 10 packages that are seamlessly supposed to work together. Therefore,
**easystats** is supported (meaning each package is thoroughly tested) on the
**current R** version, the future one (**R-devel**) and the **previous four**
versions. This is the same as the [version policy of `tidyverse`
ecosystem](https://www.tidyverse.org/blog/2019/04/r-version-support/).

Supposing R updates keep happening once a year in Spring, this table shows the 
corresponding latest R version and the oldest supported version:

| Year | Latest R version | Oldest supported R version |
|------|------------------|----------------------------|
| 2024 | 4.4              | 4.0                        |
| 2025 | 4.5              | 4.1                        |
| 2026 | 4.6              | 4.2                        |
| 2027 | 4.7              | 4.3                        |

> | 2028 | 4.8              | 4.4                        |

Reviewer: I'm not sure if there will be a version 4.8, but this table is hypothetical, I know. But then I would rather add an example that shows a major version change. E.g., what is the oldest version we support when R 5.0 or R 5.1 is released?

Reviewer 2: Hmm, I don't think it would be a good idea to use R 5.0; makes it seem like we know something that even the R core doesn't know 😆 

Rather, we can illustrate the support behaviour when major R version changes using the jump from R 3.6 to R 4.0.

Reviewer: True ;-)
But do we know that there ever will be a version 4.8? We didn't had 3.8 either.

Reviewer 2: Minor versions chosen are known to change prior to releases. So nobody has hard expectations about them. 

A change in major version is a big deal for everyone.

## Implications

The implications are simple:

> * if you are a user and your R version is below the oldest supported one, you

Reviewer 2: Nit: Since both of these unordered list items are no longer standalone clauses, can you also capitalize the first letter in the sentence.

  must update it to use the **easystats** packages. Alternatively, you can
  also download a version of the package that is compatible with your R version
  from the [CRAN archive](https://cran.r-project.org/src/contrib/Archive/) and

>   then build it locally.

Reviewer 2: Our packages contain no complicated code.Should it still be "build" (in its non-technical sense)?

---
title: "R version support"
output: 
  rmarkdown::html_vignette:
    toc: true
vignette: >
  \usepackage[utf8]{inputenc}
  %\VignetteIndexEntry{R version support}
  %\VignetteEngine{knitr::rmarkdown}
---

R evolves every year. Each update brings many bug fixes and performance 
improvements, but also sometimes changes the default value of some parameters
(such as switching from `stringsAsFactors = TRUE` to `FALSE` in R 4.0.0).

While there is benefit in supporting older R versions, doing so adds 
maintenance burden, especially in a large ecosystem like **easystats** that

> contains 10 packages that are seamlessly supposed to work together. Therefore,

Reviewer 3: ```suggestion
contains 10 packages that are supposed to seamlessly work together. Therefore,
```

**easystats** is supported (meaning each package is thoroughly tested) on the
**current R** version, the future one (**R-devel**) and the **previous four**
versions. This is the same as the [version policy of `tidyverse`
ecosystem](https://www.tidyverse.org/blog/2019/04/r-version-support/).

> Supposing R updates keep happening once a year in Spring, this table shows the 

Reviewer 3: ```suggestion
The [current planned release schedule for large R updates](https://developer.r-project.org/) is once a year in Spring. Accordingly, this table shows the 
```

corresponding latest R version and the oldest supported version:

| Year | Latest R version | Oldest supported R version |
|------|------------------|----------------------------|
| 2023 | 4.3              | 3.6                        |
| 2024 | 4.4              | 4.0                        |
| 2025 | 4.5              | 4.1                        |
| 2026 | 4.6              | 4.2                        |
| 2027 | 4.7              | 4.3                        |
| 2028 | 4.8              | 4.4                        |

> 

Reviewer 3: ```suggestion

(Of course, if the R release schedule changes or R releases a "5.0" version, the numbers in the table will change.)
```

## Implications

The implications are simple:

* If you are a user and your R version is below the oldest supported one, you
  must update it to use the **easystats** packages. Alternatively, you can
  also download a version of the package that is compatible with your R version
  from the [CRAN archive](https://cran.r-project.org/src/contrib/Archive/) and

>   then install it locally. Finally, the package `remotes` gives a way to install

Reviewer 2: For better pkgdown hyperlinking:

```suggestion
  then install it locally. Finally, the package `{remotes}` gives a way to install
```

  an old version of a package. For instance, 
  `remotes::install_version("insight", version = "0.10.0")` installs a version

>   of `insight` that is compatible with R 3.5.0.

Reviewer 2: ```suggestion
  of `{insight}` that is compatible with R 3.5.0.
```

---
title: "R version support"
output: 
  rmarkdown::html_vignette:
    toc: true
vignette: >
  \usepackage[utf8]{inputenc}
  %\VignetteIndexEntry{R version support}
  %\VignetteEngine{knitr::rmarkdown}
---

R evolves every year. Each update brings many bug fixes and performance 
improvements, but also sometimes changes the default value of some parameters
(such as switching from `stringsAsFactors = TRUE` to `FALSE` in R 4.0.0).

While there is benefit in supporting older R versions, doing so adds 
maintenance burden, especially in a large ecosystem like **easystats** that
contains 10 packages that are supposed to seamlessly work together. Therefore,
**easystats** is supported (meaning each package is thoroughly tested) on the
**current R** version, the future one (**R-devel**) and the **previous four**

> versions. This is the same as the [version policy of `tidyverse`

Reviewer: Do we need a reference to the tidyverse to justify our decision? I'd say no.

Reviewer 2: We are not citing them to justify our decision but rather to highlight that we are not alone in having such R version support restrictions. 

Or at least that's how I read it.

Reviewer: SO, we're two out of ...? Still sounds like the gray eminence to follow.

Reviewer 2: It's fine by me to get rid of it.

---
title: "R version support"
output: 
  rmarkdown::html_vignette:
    toc: true
vignette: >
  \usepackage[utf8]{inputenc}
  %\VignetteIndexEntry{R version support}
  %\VignetteEngine{knitr::rmarkdown}
---

> ```{r message=FALSE, warning=FALSE, include=FALSE}

Reviewer 2: There is no code here, so you can get rid of this setup chunk.

options(
  knitr.kable.NA = "",
  digits = 2
)

knitr::opts_chunk$set(
  comment = ">",
  dpi = 450,
  message = FALSE,
  warning = FALSE
)
```

R evolves every year. Each update brings many bug fixes and performance 
improvements, but also sometimes changes the default value of some parameters
(such as switching from `stringsAsFactors = TRUE` to `FALSE` in R 4.0.0).

> While there is benefit in supporting older R versions, doing so adds some 
> maintenance burden, especially in a large ecosystem like **easystats** that
> contains 10 packages. Therefore, **easystats** is supported (meaning
> thoroughly tested) on the latest R version, the future one (R-devel) and the
> previous 4 versions. This corresponds to the [version policy of the
> `tidyverse`](https://www.tidyverse.org/blog/2019/04/r-version-support/).

Reviewer 2: ```suggestion
While there is benefit in supporting older R versions, doing so adds 
maintenance burden, especially in a large ecosystem like **easystats** that
contains 10 packages that are seamlessly supposed to work together. Therefore, **easystats** is supported (meaning thoroughly testing) each package on the **current R** version, the future one (**R-devel**) and the
**previous four** versions. This is the same as the [version policy of `tidyverse` ecosystem(https://www.tidyverse.org/blog/2019/04/r-version-support/).
```

Supposing R updates keep happening once a year in Spring, this table shows the 
corresponding latest R version and the oldest supported version:

> | Year | Latest R version | Oldest supported R version |

Reviewer 2: Can you please fix the alignment here?

|------|----------------|--------------------------|
| 2024 | 4.4            | 4.0                      |
| 2025 | 4.5            | 4.1                      |
| 2026 | 4.6            | 4.2                      |
| 2027 | 4.7            | 4.3                      |
| 2028 | 4.8            | 4.4                      |

## Implications

The implications are simple:

* if you are a user and your R version is below the oldest supported one, you

>   must update it to use the **easystats** packages;

Reviewer 2: Also mention the possibility of getting older legacy-R compatible package versions from the CRAN archive.

* if you are a developer and your package depends on one of the **easystats**
  packages, the oldest version that your package supports corresponds to the 

>   oldest version supported by **easystats**.

Reviewer 2: And the `DESCRIPTION` should be updated to reflect this change.
